validations/src/validations.rs

```diff
@@ -367,10 +368,50 @@ pub fn validate_tally_transaction<'a>(
     let tally_stage = &dr_output.data_request.consensus;
 
     validate_consensus(&reveals, &miner_tally, tally_stage)?;
+    validate_tally_outputs(&dr_state, &ta_tx)?;
 
     Ok((ta_tx.outputs.iter().collect(), dr_output.tally_fee))
 }
 
+pub fn validate_tally_outputs(
+    dr_state: &DataRequestState,
+    ta_tx: &TallyTransaction,
+) -> Result<(), failure::Error> {
+    // Expected honest witnesses is tally outputs - 1, which would be
+    // the value transfer output related to the tally change.
+    // In case of tally change doesn't exist and all the outputs are
+    // honest witnesses rewarded, the variable is skipped by the condition
+    // that check if the pkh is equal to de data request pkh.
+    let honest_witnesses = (ta_tx.outputs.len() - 1) as u16;
+    let reveal_reward = calculate_dr_vt_reward(&dr_state.data_request);
+    for output in &ta_tx.outputs {
+        if output.pkh == dr_state.data_request.pkh {
+            let expected_tally_change =
+                reveal_reward * u64::from(dr_state.data_request.witnesses - honest_witnesses);
+            if expected_tally_change != output.value {
+                Err(TransactionError::InvalidTallyChange {
+                    change: output.value,
+                    expected_change: expected_tally_change,
+                })?
+            }
+        } else {
+            let reveal = dr_state.info.reveals.get(&output.pkh);
+
+            match reveal {
+                Some(r) => {
+                    if !true_revealer(&r, &ta_tx.tally) {
+                        Err(TransactionError::RevealDishonest)?
+                    }
```

Comment:
If it is a true revealer, check that the value of the value transfer output is equal to the witness reward

Reply:
You should also check that all the revealers were rewarded, and each of them was rewarded once and only once.